src/index.js

```diff
@@ -182,6 +216,69 @@ class Dropzone extends React.Component {
     this.fileInputEl.click();
   }
 
+  [walkDirectory](directory, callback) {
```

Comment:
Since this isn't using `this` it should not be the instance method and can be static. I would move all this directory related code to a separate module (file). Plus we will need tests to merge this one.